pkg/reconciler/route/route.go

```diff
@@ -314,16 +316,22 @@ func (c *Reconciler) tls(ctx context.Context, host string, r *v1alpha1.Route, tr
 		}
 	}
 
+	allCerts, _ := c.certificateLister.Certificates(r.Namespace).List(kubelabels.NewSelector())
```

Comment:
I am not sure if we want to look up all of the certs in a namespace.

We use the suffix of wildcard DNS name as the wildcard certificate name, and consider it as the "contract" between namespace controller and route controller, which means route controller should follow the "contract" to only look for the certificate whose name is the suffix of (major) Route DNS name.
If there is no such cert, this means namespace controller does not create a wildcard cert needed by the Route. Then Route controller should create one for the Route itself.

WDYT?

Reply:
Thinking out loud, what if we simply labeled wildcard certificates with the original domain for which they are a wildcard (pre-templating). This way we will enumerate the wildcards for the same domain entry that we matched against, and then verify that the covered domains include out own?

e.g.
```
metadata:
  labels:
    networking.knative.dev/wildcardDomain: example.com
```

Reply:
Does the `example.com` in the label mean that the wildcard domain is `*.example.com`?

I just want to make sure I understand the benefit of using label selector and enumerating the wildcards in this case. Isn't the mapping btw wildcard cert and the label is 1:1 (i.e. each cert has unique domain label)?

Reply:
We'll only ever have one wildcard certificate per namespace. Can we just use a selector like `networking.knative.dev/wildcardDomain=true`?

Reply:
I don't think one wildcard per namespace is what we want in the end-state.  e.g. I could see folks wanting a wildcard for each of the domains they have configured.

I don't think we need to handle more than this for now, but I think we should design the contract in such a way that the contract between these two controllers supports that.

Put differently: can we make it so that if/when we make `nscert` create multiple wildcards per namespace (1:1 with domains) we don't need a change to either the ingress controllers or the contract?

Reply:
It should work like that now that the certs get labeled. Anything that follows the pattern in the route reconciler (get the labeled certs for a domain then match the host with a DNSName) will work no matter how many certs get labeled.